src/target/source/source_module.cc

```diff
@@ -795,11 +796,51 @@ class MetadataSerializer : public AttrVisitor {
   std::vector<bool> is_defining_struct_;
 };
 
+namespace {
+runtime::Module CreateAotMetadataModule(runtime::metadata::Metadata aot_metadata,
+                                        bool is_c_runtime) {
+  MetadataSerializer serializer;
+  serializer.CodegenMetadata(aot_metadata);
+  std::stringstream lookup_func;
+  std::string get_c_metadata_func_name;
+
+  if (is_c_runtime == true) {
```

Comment:
maybe add:
```
// NOTE: mangling is not needed in the c++ runtime because the function name is looked-up via LibraryModule.
// TODO: unify these two approaches
```